relayer/client-tx.go

```diff
@@ -292,10 +291,31 @@ func FindMatchingClient(source, counterparty *Chain, clientState *ibctmtypes.Cli
 				continue
 			}
 
-			if IsMatchingConsensusState(consensusStateResp.ConsensusState, header.ConsensusState()) {
-				// matching client found
-				return identifiedClientState.ClientId, true
+			exportedConsState, err := clienttypes.UnpackConsensusState(consensusStateResp.ConsensusState)
+			if err != nil {
+				if source.debug {
+					source.Log(fmt.Sprintf("Error: failed to consensus state on chain %s: %v", counterparty.PathEnd.ChainID, err))
+				}
+				continue
+			}
+			existingConsensusState, ok := exportedConsState.(*ibctmtypes.ConsensusState)
+			if !ok {
+				if source.debug {
+					source.Log(fmt.Sprintf("Error:consensus state is not tendermint type on chain %s", counterparty.PathEnd.ChainID))
+				}
+				continue
 			}
+
+			if !IsMatchingConsensusState(existingConsensusState, header.ConsensusState()) {
+				continue
+			}
+
+			if existingClientState.IsExpired(existingConsensusState.Timestamp, time.Now()) {
```

Comment:
It makes sense to me to move this up to line 308 above the matching consensus state call.

NOTE: This is fine, but we may skip valid clients because we're using local time `time.Now` rather than blocktime. I don't think it's a big deal so in favor of keeping it how it is.

Reply:
yes this was my logic as well